src/realm.js

```diff
@@ -415,17 +415,21 @@ export class Realm {
     subsequentEffects[1] = pg;
     this.generator = saved_generator;
 
-    pb.forEach((val, key, m) => {
-      if (!sb.has(key)) sb.set(key, val);
-    });
-
-    pp.forEach((desc, propertyBinding, m) => {
-      if (!sp.has(propertyBinding)) sp.set(propertyBinding, desc);
-    });
-
-    po.forEach((ob, a) => {
-      so.add(ob);
-    });
+    if (pb) {
```

Comment:
Please open an issue to remind us to fix the type definition of Effects.

Reply:
Will do